ratis-server/src/main/java/org/apache/ratis/server/impl/ServerImplUtils.java

```diff
@@ -70,11 +70,15 @@ private static RaftServerProxy newRaftServer(
     return proxy;
   }
 
-  public static RaftConfiguration newRaftConfiguration(List<RaftPeer> conf, long index, List<RaftPeer> oldConf) {
+  public static RaftConfiguration newRaftConfiguration(List<RaftPeer> conf, List<RaftPeer> listener,
+      long index, List<RaftPeer> oldConf, List<RaftPeer> oldListener) {
     final RaftConfigurationImpl.Builder b = RaftConfigurationImpl.newBuilder()
-        .setConf(conf)
+        .setConf(conf, listener)
         .setLogEntryIndex(index);
-    Optional.ofNullable(oldConf).filter(p -> p.size() > 0).ifPresent(b::setOldConf);
+    if (Optional.ofNullable(oldListener).filter(p -> p.size() > 0).isPresent()
+        || Optional.ofNullable(oldConf).filter(p -> p.size() > 0).isPresent()) {
+      b.setOldConf(oldConf, oldListener);
+    }
```

Comment:
Now, we can assume that oldConf and oldListener are non-null.
```
    if (!oldConf.isEmpty() || !oldListener.isEmpty()) {
      b.setOldConf(oldConf, oldListener);
    }
```

Reply:
done